tests/components/nest/camera_sdm_test.py

```diff
@@ -0,0 +1,165 @@
+            auth=auth,
+        )
+    return await async_setup_sdm_platform(hass, PLATFORM, devices)
+
+
+async def test_no_devices(hass):
+    """Test configuration that returns no devices."""
+    await async_setup_camera(hass)
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_ineligible_device(hass):
+    """Test configuration with devices that do not support cameras."""
+    await async_setup_camera(
+        hass,
+        {
+            "sdm.devices.traits.Info": {
+                "customName": "My Camera",
+            },
+        },
+    )
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_camera_device(hass):
+    """Test a basic camera with a live stream."""
+    await async_setup_camera(
+        hass,
+        {
+            "sdm.devices.traits.Info": {
+                "customName": "My Camera",
+            },
+            "sdm.devices.traits.CameraLiveStream": {
+                "maxVideoResolution": {
+                    "width": 640,
+                    "height": 480,
+                },
+                "videoCodecs": ["H264"],
+                "audioCodecs": ["AAC"],
+            },
+        },
+    )
+
+    assert len(hass.states.async_all()) == 1
+    camera = hass.states.get("camera.my_camera")
+    assert camera is not None
+    assert camera.state == STATE_IDLE
+
+    registry = await hass.helpers.entity_registry.async_get_registry()
+    entry = registry.async_get("camera.my_camera")
+    assert entry.unique_id == "some-device-id-camera"
+    assert entry.original_name == "My Camera"
+    assert entry.domain == "camera"
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+    assert device.name == "My Camera"
+    assert device.model == "Camera"
+    assert device.identifiers == {("nest", DEVICE_ID)}
+
+
+async def test_camera_stream(hass):
+    """Test a basic camera and fetch its live stream."""
+    response = FakeResponse(
+        {
+            "results": {
+                "streamUrls": {"rtspUrl": "rtsp://some/url?auth=g.0.streamingToken"},
+                "streamExtensionToken": "g.1.extensionToken",
+                "streamToken": "g.0.streamingToken",
+                "expiresAt": "2018-01-04T18:30:00.000Z",
+            },
+        }
+    )
+    await async_setup_camera(
+        hass,
+        {
+            "sdm.devices.traits.Info": {
+                "customName": "My Camera",
+            },
+            "sdm.devices.traits.CameraLiveStream": {
+                "maxVideoResolution": {
+                    "width": 640,
+                    "height": 480,
+                },
+                "videoCodecs": ["H264"],
+                "audioCodecs": ["AAC"],
+            },
+        },
+        auth=FakeAuth(response),
+    )
+
+    assert len(hass.states.async_all()) == 1
+    cam = hass.states.get("camera.my_camera")
+    assert cam is not None
+    assert cam.state == STATE_IDLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)